### ISSUE TITLE
debian.ext3 Filesize

### ISSUE THREAD

**Issue by SuperThunder** [DylanHamer/DebianKindle#8]
Debian.ext3 is about 1.5GB for me, but seems to be at least 1gb of 0-padding. I haven't been able to move debian.ext3 to my root partition since `mv /mnt/us/debian.ext3 /` results in my KT2 running out of space. 

Should the original image be made with a lower amount of blocks (say 300) when making the empty file with dd?

**Comment by DylanHamer** [comment]
Yeah, mines 1.2GB. I'm not sure what the lowest you can go either
experiment to see what's comfortable.

**Comment by SuperThunder** [comment]
I haven't been able to get debian working yet (issues with ssh and mounting the image), but I did find some discussion on the image filesize:
https://web.archive.org/web/20110528120727/http://xkid.biz/wordpress/?p=4

It looks like the extra space is to leave room for more things installed into debian. The poster there resized his debian image with 
`dd if=/dev/zero of=debian.ext3 bs=1M count=500 oflag=append conv=notrunc
e2fsck -f debian.ext3
resize2fs debian.ext3`

I will experiment with how low you can get the image size as I get debian working.

**Comment by DylanHamer** [comment]
What issues are you having? I've probably encountered the same ones.

**Comment by SuperThunder** [comment]
On the topic of the image filesize I figured out how mounting works and it looks like the contents written to the image are only 163.8MB, so I think debian.ext3 could be a fair a bit smaller.

SSH was that I thought authorized_keys was a folder, not a file (fixed now). I also had an issue with not having any free loop devices, but it looked like 9 through 16 were actually free so I just run 
`losetup -d /dev/loop/14`
at the start of RunDebian.sh.

My main issue is that when I run `debian` to chroot into debian, I get:
[root@kindle /]# debian
'xterm-256color': unknown terminal type.
Debian Launcher 0.1.3 by Dylan Hamer
REDDIT EDITION - I have modified the
script to be a bit more self contained.

[*] Looking for Debian...
[*] Found Debian image!
[*] Mounting rootfs...
[*] Preparing Filesystem...
[*] Preparing Network...
cp: can't stat '/etc/resolv.conf': No such file or directory
[*] Starting Shell...

**FATAL: kernel too old**
[*] Unmounting rootfs...

It seems like this can be caused by the version of debian being too new for the Kindle kernel version?

**Comment by notverynerdykid** [comment]
I have encountered the same '**FATAL: kernel too old**' Problem. However, according to https://wiki.mobileread.com/wiki/Debian_Installation_on_Kindle_Touch: 

> UPDATE: Currently, "testing" is the Debian Sid release. Debootstraping it and Chrooting on a kindle will result in FATAL: KERNEL TOO OLD. To fix this replace "testing" with "wheezy" or "jessie". NOTE: Using the argument --variant=minbase will significantly reduce the size of the debian installation.


I haven't tried it yet, but would this fix the problem?

**Comment by SuperThunder** [comment]
I gave Wheezy (built with variant=minbase) a try and got this error 
[*] Looking for Debian...
[*] Found Debian image!
[*] Mounting rootfs...
[*] Preparing Filesystem...
[*] Preparing Network...
[*] Starting Shell...

**login: PAM Failure, aborting: Critical error - immediate abort**
[*] Unmounting rootfs...

I also gave Jessie (no variant=minbase) a try
[*] Looking for Debian...
[*] Found Debian image!
[*] Mounting rootfs...
[*] Preparing Filesystem...
[*] Preparing Network...
[*] Starting Shell...

**chroot: can't execute '/bin/login': No such file or directory**
[*] Unmounting rootfs...

No /bin/login has had me scratching my head. I took a look in debian.ext3 and /bin/login is definitely there.
For whatever reason the mount command on the Kindle seems to just not be actually mounting the contents of the folders of the image.

[root@kindle /bin]# cd /mnt/debian
[root@kindle debian]# ls
ls: ./mnt: Input/output error
ls: ./selinux: Input/output error
bin          dev          lib          root         sys          var
boot         etc          lost+found   run          tmp
debootstrap  home         proc         sbin         usr

When I cd into bin, lib, sys, etc there are none of the contents that should be there and are there if I mount the same .ext3 on Ubuntu.

**Comment by DylanHamer** [comment]
Tis very strange, I've never had this issue before. I'll generate an image
for you but I can't test it because I broke my Kindle.

**Comment by SuperThunder** [comment]
Oh no, permanently broken or just software issues?

I might try the whole process from scratch at some point in case something got messed up somewhere.
I wonder if chrooting into another distro like Arch or Alpine would run into the same issues.

**Comment by GNOME-IS-LIFE** [comment]
Any update guys? confused dev here.

**Comment by DaJakerBoss** [comment]
Okay @SuperThunder the Jessie variant ran all the way to the end of the Debian run script. The "/bin/login not found" chroot error most of the time has to do with issues in your debian image having the incorrect architecture. Can you just confirm real quick that you have an ARM image of debian?
